### DISCUSSION TITLE
conditional formatting not allowing me to use percent formating

### DISCUSSION THREAD

**Discussion by CorbinAutomations** [qax-os/excelize#1610] [Q&A]
so I have some code that uses conditional formatting and I ran into this problem where anywhere that had conditional formatting my percentage formatting using SetCellStyle would disappear. I tried to add this style to the conditional formatting by using    
SetConditionalformatting but this still did not work. I have no Idea why this is happening does anyone have any ideas thx in advanced

**Comment by xuri** [comment]
I have added support for creating a conditional format with number format and protection, Please try to upgrade the master branch code, and this feature will be released in the next version. For example, you can format A1:A10 cells with values between 0 to 1 with 2 places decimal percentage number format:

```go
package main

import (
    "fmt"

    "github.com/xuri/excelize/v2"
)

func main() {
    f := excelize.NewFile()
    defer func() {
        if err := f.Close(); err != nil {
            fmt.Println(err)
        }
    }()
    format, err := f.NewConditionalStyle(
        &excelize.Style{
            // Using built-in number format code at here,
            // you can also using the custom number format code
            NumFmt: 10,
        },
    )
    if err != nil {
        fmt.Println(err)
        return
    }
    // Highlight cells rule: between...
    if err := f.SetConditionalFormat("Sheet1", "A1:A10",
        []excelize.ConditionalFormatOptions{
            {
                Type:     "cell",
                Criteria: "between",
                Format:   format,
                MinValue: "0",
                MaxValue: "1",
            },
        },
    ); err != nil {
        fmt.Println(err)
        return
    }
    if err := f.SaveAs("Book1.xlsx"); err != nil {
        fmt.Println(err)
    }
}
```

**Reply by CorbinAutomations** [reply]
that is fantastic. I have not been using excelize long but I have to say everything has been great from the product to the docs and the support. I am yet to have a negative experience with it. It really has helped me as a beginner to learn both golang and excelize be able to create something useful and practice all of golangs features. I have been programming for 2 weeks and I am already using it for 3 separate projects including a cloud native product that will turn a task that is costing my dads work 19 hours a day into something that takes 2.9 seconds I could not recommend it enough.

**Reply by rashi-j** [reply]
Hey @xuri is it possible to have conditional formatting for drop-down values currently? Like if user uses one of the drop-down values in a cell, the fill and font should have different formatting. And each value comes with its format. Something like below is what I want to achieve:

```
format, err := f.NewConditionalStyle(
		&excelize.Style{
			Fill: excelize.Fill{Type: "pattern", Color: []string{"F0FFFC"}, Pattern: 1},
			Font: &excelize.Font{Color: "069A7A"},
		},
	)
	if err != nil {
		fmt.Println(err)
		return
	}
	// Highlight each cell with value draft
	if err := f.SetConditionalFormat("Sheet1", "A1:A10",
		[]excelize.ConditionalFormatOptions{
			{
				Type:     "cell",
				Criteria: "equal to",
				Format:   format,
				Value:    "draft",
			},
		},
	); err != nil {
		fmt.Println(err)
		return
	}

**Reply by xuri** [reply]
Yes, please add a double quote for the "draft", and if not,  which represents the defined names. For example:


```go
package main

import (
    "fmt"

    "github.com/xuri/excelize/v2"
)

func main() {
    f := excelize.NewFile()
    defer func() {
        if err := f.Close(); err != nil {
            fmt.Println(err)
        }
    }()
    format, err := f.NewConditionalStyle(
        &excelize.Style{
            Fill: excelize.Fill{Type: "pattern", Color: []string{"F0FFFC"}, Pattern: 1},
            Font: &excelize.Font{Color: "069A7A"},
        },
    )
    if err != nil {
        fmt.Println(err)
        return
    }
    // Highlight each cell with value draft
    if err := f.SetConditionalFormat("Sheet1", "A1:A10",
        []excelize.ConditionalFormatOptions{
            {
                Type:     "cell",
                Criteria: "equal to",
                Format:   format,
                Value:    "\"draft\"",
            },
        },
    ); err != nil {
        fmt.Println(err)
        return
    }

    dv := excelize.NewDataValidation(true)
    dv.Sqref = "A1:A10"
    if err := dv.SetDropList([]string{"draft", "another value"}); err != nil {
        fmt.Println(err)
        return
    }
    if err := f.AddDataValidation("Sheet1", dv); err != nil {
        fmt.Println(err)
        return
    }

    if err := f.SaveAs("Book1.xlsx"); err != nil {
        fmt.Println(err)
    }
}
```

**Reply by rashi-j** [reply]
Thank you @xuri . That worked perfectly!
